diff --git a/src/http/modules/ngx_http_grpc_module.c b/src/http/modules/ngx_http_grpc_module.c
index e7726f3..3d0f958 100644
--- a/src/http/modules/ngx_http_grpc_module.c
+++ b/src/http/modules/ngx_http_grpc_module.c
@@ -10,6 +10,11 @@
 #include <ngx_http.h>
 
 
+#if (NGX_HTTP_PROXY_VAR_SET)
+#include "ngx_http_proxy_var_set_module.h"
+#endif
+
+
 typedef struct {
     ngx_array_t               *flushes;
     ngx_array_t               *lengths;
@@ -1928,6 +1933,12 @@ ngx_http_grpc_process_header(ngx_http_request_t *r)
                     }
                 }
 
+#if (NGX_HTTP_PROXY_VAR_SET)
+            if (ngx_http_grpc_var_set_handler(r) != NGX_OK) {
+                return NGX_ERROR;
+            }
+#endif
+
                 return NGX_OK;
             }
 
diff --git a/src/http/modules/ngx_http_proxy_module.c b/src/http/modules/ngx_http_proxy_module.c
index c69a476..1468f20 100644
--- a/src/http/modules/ngx_http_proxy_module.c
+++ b/src/http/modules/ngx_http_proxy_module.c
@@ -10,6 +10,11 @@
 #include <ngx_http.h>
 
 
+#if (NGX_HTTP_PROXY_VAR_SET)
+#include "ngx_http_proxy_var_set_module.h"
+#endif
+
+
 #define  NGX_HTTP_PROXY_COOKIE_SECURE           0x0001
 #define  NGX_HTTP_PROXY_COOKIE_SECURE_ON        0x0002
 #define  NGX_HTTP_PROXY_COOKIE_SECURE_OFF       0x0004
@@ -2020,6 +2025,12 @@ ngx_http_proxy_process_header(ngx_http_request_t *r)
                 }
             }
 
+#if (NGX_HTTP_PROXY_VAR_SET)
+            if (ngx_http_proxy_var_set_handler(r) != NGX_OK) {
+                return NGX_ERROR;
+            }
+#endif
+
             return NGX_OK;
         }
 
